climada/util/coordinates.py

```diff
@@ -1821,6 +1821,140 @@ def apply_box(df_exp):
     }
     return raster_out, meta
 
+def reproject_2d_grid(data, meta_in, meta_out, res_arcsec_out=None,
+                      global_origins=(-180.0, 90.0), crs_out=None, resampling=None,
+                      conserve=None, buffer=0):
+    """
+    Reproject (up- or downsample) 2D np.ndarray to a grid using rasterio.
+    The destination (target) grid is defined from target_meta and optionally
+    target_res_arcsec, global_origins, and target_crs.
+    This function ensures that reprojected data with the same resolution and
+    global_origins are on the same global grid, i.e., no offset between 
+    destination grid points for different input grids that are projected to
+    the same resolution.
+
+    Parameters
+    ----------
+    data : np.ndarray
+        2D-array containing the values to be reprojected
+    meta_in : dict
+        meta data dictionary of input grid
+        Required fields in meta are 'dtype,', 'width', 'height', 'crs', 'transform'.
+        Example:
+            {'dtype': 'float32',
+             'width': 2702,
+             'height': 1939,
+             'crs': CRS.from_epsg(4326),
+             'transform': Affine(0.00833333333333333, 0.0, -18.175000000000068,
+                                 0.0, -0.00833333333333333, 43.79999999999993)}
+    meta_out : dict
+        meta data dictionary of target grid, same structure as meta_in.
+        meta_out defines the target (i.e. destination) grid.
+    res_arcsec_out : int (optional)
+        target resolution in arcsec, overrules resolution in meta_out['transform'].
+        The default is None, i.e. same resolution as target grid defined in meta_out.
+    global_origins : tuple with two numbers (lat, lon) (optional)
+        global lon and lat origins as basis for destination grid.
+        Only required if target_res_arcsec different than resolution in meta_out.
+        The default is (-180.0, 90).
+    crs_out : rasterio.crs.CRS
+        destination CRS
+        The default is crs provided in meta_out.
+    resampling : resampling function (optional)
+        The default is rasterio.warp.Resampling.bilinear
```

Comment:
I would use `Resampling.nearest` as a default because that's also the default in the origina `rasterio.warp.reproject`.